MLS.Agent.Tools/AddPackageResult.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MLS.Agent.Tools
+{
+    public partial class Dotnet
+    {
+        public class AddPackageResult : CommandLineResult
+        {
+            
+            public string InstalledVersion { get; }
+            public IEnumerable<string> DetailedErrors { get; }
+
+            public AddPackageResult(int exitCode, IReadOnlyCollection<string> output = null, IReadOnlyCollection<string> error = null) : base(exitCode, output, error)
+            {
+                if (exitCode == 0)
+                {
+                    InstalledVersion = GetInstalledVersion(output);
+                }
+
+                DetailedErrors = GetDetailedErrors(output, error);
+            }
+
+            private IEnumerable<string> GetDetailedErrors(IReadOnlyCollection<string> output, IReadOnlyCollection<string> error)
+            {
+                return output.Concat(error).Where(l => l.StartsWith("error:"));
+            }
+
+            private string GetInstalledVersion(IReadOnlyCollection<string> output)
+            {
+                foreach (var line in output)
+                {
+                    if (line.StartsWith("info : PackageReference for package"))
+                    {
+                        // Successful installation will print a message like
+                        // @"info : PackageReference for package 'xunit' version '2.4.1' added to file 'E:\testdll\testdll.csproj'.";
+                        var quote1 = line.IndexOf('\'');
+                        if (quote1 == -1)
+                        {
+                            continue;
+                        }
+
+                        var quote2 = line.IndexOf('\'', quote1 + 1);
```

Comment:
so it is 0 or 4 ```\``` char always?

Reply:
Can't have ' in package name or version number...